code/controllers/subsystem/dbcore.dm

```diff
@@ -90,17 +76,27 @@ SUBSYSTEM_DEF(dbcore)
 	var/address = CONFIG_GET(string/address)
 	var/port = CONFIG_GET(number/port)
 
-	_dm_db_connect(_db_con, "dbi:mysql:[db]:[address]:[port]", user, pass, Default_Cursor, null)
-	. = IsConnected()
+	connection = new /datum/BSQL_Connection(BSQL_CONNECTION_TYPE_MARIADB)

```

Comment:
I was going to suggest making this asyncable but i suppose the primary issue (long&blocking connection timeouts) could be fixed by just using a sane connection timeout. what is bsql using?

Reply:
uhh whatever the default is for mariadb. (I've been meaning to add that as a parameter, I'll get on that)